### PR TITLE
chore: Codemirror semantic styles.

### PR DESCRIPTION
**Pull request by richburdon**
NOTE: Given the late hour I have dark mode reader on so I haven't checked how all the specific colors look yet! :).

- [x] Replaced getToken with semantic colors.
- [x] Moved markdown styles out of base theme.
- [x] Removed light/dark selectors from styles and converted to regular theme (not base theme).
- [x] Updated styles for search panel. 
- [x] Toolbar search button. 

Questions (@thure):
- how to deal with fonts?
- how to normalize inputs/buttons, etc. (e.g., .cm-panel).

Screenshot 2024-09-13 at 3 26 47 PM

### CODE REVIEW

File: packages/ui/react-ui-theme/src/config/tokens.ts [hunk 1]
```diff
@@ -301,6 +302,49 @@ const semanticColors = {
       light: ['red', 700],
       dark: ['red', 300],
     },
+
+    //
+    // Codemirror
```
Comment (wittjosiah) [inline packages/ui/react-ui-theme/src/config/tokens.ts:307]: I imagine that we may want to factor many of these out from "cm specific styles" to "general theme styles" over time, but given they're specifically being used for cm editor right now this makes that quite clear 👍

File: packages/plugins/plugin-mermaid/src/extensions/mermaid.ts [hunk 2]
```diff
@@ -155,6 +152,6 @@ const styles = EditorView.baseTheme({
   // },
   '& .cm-mermaid-error': {
     display: 'inline-block',
-    color: getToken('extend.colors.red.500'),
+    color: 'var(--dx-error)',
```
Comment (thure) [inline packages/plugins/plugin-mermaid/src/extensions/mermaid.ts:155]: I meant to remove/deduplicate/qualify this `error` token with the rest in `messageValence.ts` — I think the scope of this PR could, maybe should, be extended to handle that too.

File: packages/ui/react-ui-theme/src/config/tokens.ts (hunk 1 above)
Comment (thure) [reply packages/ui/react-ui-theme/src/config/tokens.ts:307]: Yeah this is an interesting use-case… maybe in a later iteration `react-ui-editor` or MarkdownPlugin can export these sememes and composer could provide it to the Vite config.

File: packages/plugins/plugin-mermaid/src/extensions/mermaid.ts (hunk 2 above)
Comment (thure) [reply packages/plugins/plugin-mermaid/src/extensions/mermaid.ts:155]: Maybe not, though, this PR seems to just address CM and that’s fine, can remove the bare `error` token in another pass that handles `messageValence`.

**Pull request merged**